### DISCUSSION TITLE
Add logo to repository

### DISCUSSION THREAD

**Discussion by fabiobeoni** [community/community#22019] [New to GitHub]
<p>Hello,</p>
<p>anyone knows if there is a way to add a logo to a repository?</p>
<p>I’m experiencing a strange “social” sharing: someone on LinkedIn shares one of my repos, and the sharing message linking the repo shows my github profile picture… sounds strange, it would be better to have there a logo related to the project.</p>
<p>Thanks! F.</p>

**Comment by mpboom** [comment]
<p>I don’t blieve that is possible.</p>

**Reply by knx555** [reply]
It is possible, Github just decided to remove the functions for us small developers who can't or don't like to pay for a pricy enterprise account.

**Comment by Fatemehshirazi** [comment]
<p>Hi,</p>
<p>I have sett some free guide on the internet, where show you how you can fix your own logo. Then you can use that</p>
<p>logo as your Github profile picture. Do a google search.</p>
<p>/Fatemeh</p>

**Reply by knx555** [reply]
Off topic and BS! We are talking about the image for a repository. You should read before you post.

**Comment by DimitriPapadopoulos** [comment]
<p>While it is possible to add a logo to a GitHub profile, as far as I know one <a href="https://webapps.stackexchange.com/questions/54045/can-i-add-an-image-as-a-logo-of-a-github-repository#54058" rel="nofollow noopener">cannot add a logo to a repository</a> - except by adding an image to the README file. The question was about repositories.</p>

**Reply by knx555** [reply]
Correct, we are talking about repos and not social media.

**Comment by elmar-chen** [comment]
<p>Is there anyway to propose a feature request to github for this?</p>

**Reply by knx555** [reply]
No, Github is stoichly ignoring it's customers.

**Comment by mpboom** [comment]
<p>Yes. GitHub Support.</p>

**Reply by knx555** [reply]
No, Github is stoichly ignoring it's customers.

**Comment by prahladyeri** [comment]
<p>Interestingly, this page tops the search result for <code>github repo logo</code> which explains the intensity of this problem! I’m also facing the same issue as you, if I post a link to my github repo, my profilie picture appears as the featured image which is irrelevant and pointless.</p>
<p>This should be trivial to implement. For example, you can take a file like <code>logo.png</code> or <code>favicon.png</code> from the repo’s root directory and make it as the featured image, isn’t it? Gitlab does the same to the best of my knowledge.</p>
<p>Hoping this will be resolved soon my Github support!</p>

**Reply by knx555** [reply]
No, Github is stoichly ignoring it's customers.

**Comment by mpboom** [comment, marked as answer]
<p>Hi <a class="mention" href="/prahladyeri">@prahladyeri</a> ,</p>
<p>This issue already has been resolved! See the solution below. This is also interesting for the other contributors to this conversation: <a class="mention" href="/fabiobeoni">@fabiobeoni</a> , <a class="mention" href="/dimitripapadopoulos">@dimitripapadopoulos</a> , <a class="mention" href="/fatemehshirazi">@fatemehshirazi</a> .</p>
<p>You can navigate to your repo &gt;  <strong>Settings </strong> &gt;  <strong>Options</strong> &gt; and add an image under  <strong>Social media preview</strong>. Good luck!</p>

**Reply by ahmedmansoor** [reply]
Actually, now it's **Settings** > **Social preview**.
source: [Customizing your repository's social media preview](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/customizing-your-repositorys-social-media-preview)

**Reply by janpfeifer** [reply]
This would be great if it worked ... I changed the image under Settings > Social preview, and it still displays my profile picture for the repository. Any ideas ?

https://github.com/janpfeifer/gonb

Thanks!

**Reply by mpboom** [reply]
@janpfeifer probably just a patience will go a long way. Since Google probably already indexed the "old" profile picture image, it'll take a while for it to disappear. Same goes for any messaging/social platforms that already discovered the old thumbnail.

**Reply by erickveil** [reply]
I think we all have a miscommunication here. 

@janpfeifer is expecting the _icon_ for the repository here on github to change, I think. Behavior you would see on, for example, GitLabs.
But the solution here is just for creating a sort of "social media preview" image. You're only going to see it if you, say, post a link to your repository on Reddit or Xitter. 

I don't know of a way to change the icon that appears for a specific repository.

**Reply by knx555** [reply]
Off topic and BS! We are talking about the image for a repository. You should read before you post.

**Comment by DimitriPapadopoulos** [comment]
<p>Ah indeed, this is documented in <a href="https://help.github.com/en/articles/customizing-your-repositorys-social-media-preview" rel="nofollow noopener">Customizing your repository’s social media preview</a>.</p>
<p>I have added a logo to one of my repositories. I’m not much in social networks. When I send an SMS with a link to the repository, no preview of any kind appears on Android. Is this very specific to some social networks? Where can I find documentation on the <strong>social media preview</strong> specification, if there is one?</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post.

**Comment by mpboom** [comment]
<p>Great that you posted a link to the help article! I don’t believe there is more specific information availible. I thinks GitHub makes use of <a href="http://ogp.me/" rel="nofollow noopener">http://ogp.me/</a> , which transfers this kind of info trough HTML  <strong>meta</strong> tags.</p>
<p>It is an open protocol and as far as I know it’s supported on most social media and messaging services.</p>
<p>As in your case, I just think your Android SMS application doesn’t allow for previews at all. If it were to allow any preview at all, it would show your profile picture as an image  <strong>until</strong> you uploaded a custom image. So if you aren’t seeing any image, I don’t think your client supports it.</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post.

**Comment by prahladyeri** [comment]
<p>Thanks, that’s helpful!</p>

**Reply by knx555** [reply]
Which part of the off topic part is helpful? We are talking about the image for a repository. You should read before you post.

**Comment by Shailaja-Gupta** [comment]
<p>Hi I followed all steps and added logo to reporsitory. However still when I put my repository link on linkedin, I am getting no image. The image is reflecting though on my Settings&gt;Options&gt;Social preview. What to do? I am using laptop to view</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by prahladyeri** [comment]
<p>Does it show up on twitter cards preview? Here is the test link for twitter cards preview, if it shows up here then there is no need to worry as it could be Linkedin’s caching problem, it’ll be sorted after some time when linkedin knows that your site’s logo is updated.</p>
<p><a href="https://cards-dev.twitter.com/validator" rel="nofollow noopener">https://cards-dev.twitter.com/validator</a></p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by anmolter** [comment]
<p>Hi,<br>
when I put the link to my repository into the card validator, it shows the social media image that I have added. However, when I type the link to my repository into a tweet, it shows my profile image. Is there a specific method that I need to use to link to my repository via Twitter?</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by MaximilianKohler** [comment]
<p>For the Twitter validator I’m getting:</p>
<pre><code class="lang-auto">Unable to render Card preview
Log
ERROR: Fetching the page failed because it's denied by robots.txt.
</code></pre>
<p>And Facebook is still showing my profile picture.</p>
<p>LinkedIn post inspector <a href="https://www.linkedin.com/post-inspector/inspect" rel="noopener nofollow ugc">https://www.linkedin.com/post-inspector/inspect</a> is showing it correctly.</p>
<p>EDIT: I got the Facebook preview to work via <a href="https://developers.facebook.com/tools/debug/" rel="noopener nofollow ugc">https://developers.facebook.com/tools/debug/</a> - “scrape again”. Even though the debug tool said it had been scraped recently already.</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by EdoardoCocconi** [comment]
<p>maybe try to make a new link to your repo with tinyurl. Linkedin has probably saved your old pic already</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by EdoardoCocconi** [comment]
<p>Actually tinyurl didn’t work fro me and the picture got back to my profile picture. If you don’t want to wait you can change the name of the repos so you don’t have to wait for linkedin to catch up</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by leoheck** [comment]
<p>I found this thread, having the same/similar idea.</p>
<p>My idea is that the repo could have a logo. if there is no place to show this logo, it could be used instead of the owner logo when referencing the repo.</p>

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about avatar.

**Comment by ris58h** [comment]
Why Social Preview has to be rectangular? Most logos are square.

**Reply by mpboom** [reply]
Square logos will get cropped automatically; by being able to upload a rectangular logo you are sure to support all platforms (also the ones that offer rectangular logos), and not just platforms that display square logos.

**Reply by ris58h** [reply]
You could swap words square and rectangular in your statement and the result would be the same - some platforms support it but some not.

Logos shouldn't be cropped in the first place and most projects already have square logos.

**Reply by mpboom** [reply]
No, you can not ‘swap words’. A rectangular logo contains more information than a square one. You can cut off part of a rectangular logo to make it square, but not the other way around. Note that GitHub isn’t doing the cropping; the individual implementations of the preview are. GitHub provides the full inage to clients, who then may or may not crop the rectangular logo into a square one.

**Reply by mpboom** [reply]
But yeah, if your project has a square logo, and you’re set on using that, I’d recommend just applying a background to it with an even color.

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about social media.

**Comment by thadguidry** [comment]
Adding Images or SVG's to Profiles or README's can now be done with the `<picture>` element as explained in GitHub Official Docs
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/quickstart-for-writing-on-github

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post. It is not about Profiles or READMEs.

**Comment by thadguidry** [comment]
There's a few things that GitHub could do that can improve things overall:
1. Allow uploading not only the existing Social Media card in Settings, but also allowing upload of a Logo which would populate the HTML header with `og:image` as well as Schema.org/image with below.
2. alternatively/additionally adding a bit of JSON-LD with [schema.org/SoftwareSourceCode](https://schema.org/SoftwareSourceCode) and using its [`image`](https://schema.org/image) property.
    - Incidentally, **all repo's already** have microdata applied with:
        `<div itemscope itemtype="http://schema.org/SoftwareSourceCode" class>`
         but many Schema.org properties are not utilized such as:
          `image`, `about`, `description`, `license`, `maintainer`, `discussionUrl`, `contributor`, `programmingLanguage`, `codeRepository`, `sourceOrganization`, etc.
    - It would be best to utilize a JSON-LD script and populate it with relevant Schema.org properties and remove the microdata old ways.

### As one of the contributors to Schema.org, I'd be willing to help GitHub and discuss the small improvements necessary, for FREE.

**Reply by knx555** [reply]
Off topic! We are talking about the image for a repository. You should read before you post.

**Comment by Jay-Karia** [comment]
How can we change the repo icon
![ghRepoIcon](https://github.com/user-attachments/assets/2f413030-510d-4948-929e-43d1fa73e7b0)

To something like this
![withIcon](https://github.com/user-attachments/assets/2eb21598-aa27-459b-b3a1-6bbf700eff98)

**Reply by bo1ta** [reply]
> How can we change the repo icon ![ghRepoIcon](https://private-user-images.githubusercontent.com/86977383/366919196-2f413030-510d-4948-929e-43d1fa73e7b0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1MzIxMDEsIm5iZiI6MTcyNzUzMTgwMSwicGF0aCI6Ii84Njk3NzM4My8zNjY5MTkxOTYtMmY0MTMwMzAtNTEwZC00OTQ4LTkyOWUtNDNkMWZhNzNlN2IwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI4VDEzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MmMyZjQxOGI1Nzg5MTBmMDAxOWVlNWNmMDcwNmI2ZDFjZTUyMTVhNjY4ZTBkMWExMzk2ODgzY2IwMTgxYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4y-vYsuXnR1D5dgUE4jDdWWxlJEFY9W8hTkbyaFDevo)
> 
> To something like this ![withIcon](https://private-user-images.githubusercontent.com/86977383/366919284-2eb21598-aa27-459b-b3a1-6bbf700eff98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1MzIxMDEsIm5iZiI6MTcyNzUzMTgwMSwicGF0aCI6Ii84Njk3NzM4My8zNjY5MTkyODQtMmViMjE1OTgtYWEyNy00NTliLWIzYTEtNmJiZjcwMGVmZjk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI4VDEzNTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2M2ZmNzlkODIzNjE1OTk3MGNhY2JmNjFiYmI2MjU5MzYxNDYwYzc1ZTM2YjdlODA5MDNhNDRlMDk0NjMyMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3Z80HqNAx1MmERfAiTxOEbZglC3sUS7RjglAI25rb9c)

I think that's just for organizations

**Reply by knx555** [reply]
It's a non-sense limitation for paying customers who don't have a pricy enterprise account. And Github displays it very well and often, that they don't care a single bit for small developers. They hate us, but they like taking our money.

**Reply by Rudra-G-23** [reply]
Can we do this?
for the repo?

<img width="615" height="307" alt="image" src="https://github.com/user-attachments/assets/01238495-6d05-4065-a958-c6386475b31c" />

**Comment by dskvr** [comment]
Why are people talking about "Social Media Preview" the question clearly asks "Add logo to Repository"

<img width="50" alt="Screen Shot 2025-03-28 at 3 53 33 PM" src="https://github.com/user-attachments/assets/63a4dccd-4412-43fb-bb95-363de3cdd33b" /> <- how do we change the LOGO, I know it can be changed, but how.

**Reply by knx555** [reply]
People don't like to read anymore. They just post crap to get the feeling to be heard.

**Comment by knx555** [comment]
Why can't I set a logo on a private repository? That's a nonsense limitation!

**Reply by reactsaas** [reply]
yeah social preview to upload image only available in public repo  ;(

**Comment by knx555** [comment]
The proposed solution is a solution to another problem. We as paying customers want that feature for our repos and not the so called "social media" crap. But it's alright, I already started migrating my stuff to my own Git-Server. 

So, F you Github, if you anger us with that shitty non-sense limitation and other limitations for paying customers who don't have a pricy enterprise account, you will lose those customers.

**Reply by knx555** [reply]
It's an image. We want just a simple fucking image for our private repos. But Github pretends it's an elite feature for enterprises.

**Comment by knx555** [comment]
Since the original question was definitly NOT answered, this discussion should not be marked as "answered" or solved.

**Comment by AldeRoberge** [comment]
To anyone coming later, don't be bothered. GitHub doesn’t allow custom icons. They’ll always use your personal profile picture instead.

There’s no setting, API, or workaround to override that behavior right now. Even if you upload an image in .github/, make a logo in the README, or use GitHub Pages, the actual repo icon shown on your profile or search results will still default to your profile avatar.

**Reply by andreasmolnardev** [reply]
Apparently they have to keep making GitHub worse instead of actually adding features that.. would make sense

**Reply by AldeRoberge** [reply]
> Apparently they have to keep making GitHub worse instead of actually adding features that.. would make sense

Honestly, they could probably just ask GitHub Copilot to do it, and it would be done in a few minutes. I hope AI helps accelerate these kinds of quality-of-life features. Things that may not directly increase revenue but significantly improve the user experience.
